From d63479b24767b029b68c6e13bf3082f1329bc4bf Mon Sep 17 00:00:00 2001
From: Venkatarama Avadhani <venkatarama.avadhani@ittiam.com>
Date: Mon, 18 Dec 2017 15:57:18 +0530
Subject: [PATCH] Correcting Buffer Allocation for Shared Display

In case of shared display mode, the picture buffer allocated has to be
larger to accomodate the incorrect half pel reference to the last row in
the picture. Adding memory for the same.

Bug: 70350015
Bug: 70349694
Bug: 70349612
Bug: 70349754
Bug: 70349868
Bug: 70526352
Bug: 70350086
Test: manual
Change-Id: I41905d101093ae20ab14193c21669b8c4a24f30c
(cherry picked from commit 5c1ed47b1fc1a0f763c251d62db14f3d74cb3141)
---

diff --git a/decoder/impeg2d_api_main.c b/decoder/impeg2d_api_main.c
index 5693555..0a59310 100644
--- a/decoder/impeg2d_api_main.c
+++ b/decoder/impeg2d_api_main.c
@@ -824,7 +824,8 @@
     {
         u4_stride = ps_dec_state->u4_frm_buf_stride;
     }
-    u4_height = ((ps_dec_state->u2_frame_height + 15) >> 4) << 4;
+    u4_stride = ALIGN16(u4_stride);
+    u4_height = ALIGN32(ps_dec_state->u2_frame_height) + 9;
 
     if(ps_dec_state->i4_chromaFormat == IV_YUV_420P)
     {
